test/api/group_sets_api_test.rb

```diff
@@ -0,0 +1,44 @@
+require 'test_helper'
+require 'user'
+
+class GroupSetsApiTest < ActiveSupport::TestCase
+  include Rack::Test::Methods
+  include TestHelpers::AuthHelper
+  include TestHelpers::JsonHelper
+  include TestHelpers::TestFileHelper
+
+  def app
+    Rails.application
+  end
+
+  def test_get_all_groups_in_unit_with_authorization
+    # Create a group
+    newGroup = FactoryBot.create(:group)
+    # Obtain the unit of the group
+    newUnit = newGroup.group_set.unit
+    get with_auth_token "/api/units/#{newUnit.id}/groups",newUnit.main_convenor_user
+    
+    #Check response
+    response_keys = %w(id name)
+    last_response_body.each do | data |
+      grp = Group.find(data['id'])
+      assert_json_matches_model(data, grp, response_keys)
+      assert_equal data['name'],grp['name']
+    end
+    assert_equal 200, last_response.status
+  end  
+
+  def test_get_all_groups_in_unit_without_authorization
+    # Create a group
+    newGroup = FactoryBot.create(:group)
+    # Obtain the unit of the group
+    newUnit = newGroup.group_set.unit
+    # Create student
+    studentUser = FactoryBot.create(:user, :student)
```

Comment:
The GET call should be performed by user in the related unit, if you create a new student user, that user will be in a different unit. To make the test more logic, I suggest you should assign a student user from the above unit [`student = newUnit.active_projects.first.student`] to perform the GET.